tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -240,6 +240,59 @@ def definition_func(inout_field: gtscript.Field[float]):
                 definition_func, name=inspect.stack()[0][3], module=self.__class__.__name__
             )
 
+    def test_use_in_expr(self):
+        @gtscript.function
+        def func():
```

Comment:
Just a question outside of the scope of this PR: is it explicitly check that there are not recursive calls (or mutually recursive calls) inside function definitions?

Reply:
I opened an issue for recursive calls. As we expected, it just ends up as an infinite recursion in the frontend.